CMakeLists.txt

```diff
@@ -162,7 +160,7 @@ if(AVIF_LOCAL_LIBYUV)
     if(DEFINED ANDROID_ABI)
         set(AVIF_LIBYUV_BUILD_DIR "${AVIF_LIBYUV_BUILD_DIR}/${ANDROID_ABI}")
     endif()
-    set(LIB_FILENAME "${AVIF_LIBYUV_BUILD_DIR}/${AVIF_LIBRARY_PREFIX}yuv${AVIF_LIBRARY_SUFFIX}")
+    set(LIB_FILENAME "${AVIF_LIBYUV_BUILD_DIR}/${AVIF_LIBRARY_PREFIX}yuv${CMAKE_STATIC_LIBRARY_SUFFIX}")
```

Comment:
Yannis: Could you review this pull request? It reverts some of the changes you made in https://github.com/AOMediaCodec/libavif/pull/1169.

Reply:
It's useful whenever you want to distribute a library that provides bindings to libavif with a bundled build of libavif. For python and node, this is accomplished by shipping your extension with a shared library that has a modified RPATH, to get picked up at runtime.

I could be doing something wrong, but I have had difficulty linking against a static build of libavif. It seems that I need to combine the other static libraries using `LIB.EXE` or `ar`, or else add them to the link flags when I build the extension. And the libavif.pc file doesn't include `Libs.private` or `Requires.private`, so I can't use pkg-config to figure out what those link flags should be.

Reply:
Yannis: We can still test "everything shared" in the CI. It's just that the dependent libraries such as libaom and dav1d cannot be locally built (in the `ext` directory) and need to be installed as packages.

Frankie: I am not familiar with pkg-config, but I think libavif.pc should imitate how libpng's .pc file handles the zlib dependency (assuming libpng does it correctly):
https://github.com/glennrp/libpng/blob/libpng16/libpng.pc.in

Reply:
@fdintino, just to be sure: you were building libavif as static, with dynamic deps, installing it and using pkgconfig to use it? (not CMake)

Reply:
Closing the loop on this conversation: as I noted in a comment on the PR itself, I've opted to include the commit that fixes the static library bundling as an additional commit here.